tools/setup_perf_cluster.sh

```diff
@@ -0,0 +1,151 @@
+#! /bin/bash
+# Sets up a cluster for perf testing - GCP/GKE
+# PROJECT=istio-perf tools/setup_perf_cluster.sh
+set -e
+
+ZONE=${ZONE:-us-east4-b}
+CLUSTER_NAME=${CLUSTER_NAME:-istio-perf}
+MACHINE_TYPE=${MACHINE_TYPE:-n1-highcpu-2}
+NUM_NODES=${NUM_NODES:-6} # SvcA<->SvcB + Ingress + Pilot + Mixer + 1 extra (kube-system)
+VM_NAME=${VM_NAME:-fortio-vm}
+ISTIOCTL=${ISTIOCTL:-istioctl} # to override istioctl from outside of the path
+FORTIO_NAMESPACE=${FORTIO_NAMESPACE:-fortio}
+
+function usage() {
+    echo "usage: PROJECT=project ZONE=zone $0"
+    echo "also settable are NUM_NODES, MACHINE_TYPE, CLUSTER_NAME, VM_NAME, VM_IMAGE"
+    exit 1
+}
+
+if [[ -z "${PROJECT}" ]]; then
+  usage
+fi
+
+GCP_OPTS="--project $PROJECT --zone $ZONE"
+
+function execute() {
+  echo "### Running:" "$@"
+  "$@"
+}
+
+function create_cluster() {
+  execute gcloud container clusters create $CLUSTER_NAME $GCP_OPTS --machine-type=$MACHINE_TYPE --num-nodes=$NUM_NODES --no-enable-legacy-authorization
+}
+
+function create_vm() {
+  VM_IMAGE=${VM_IMAGE:-$(gcloud compute images list --standard-images --filter=name~ubuntu-1604-xenial --limit=1 --uri)}
+  echo "Creating VM_NAME=$VM_NAME using VM_IMAGE=$VM_IMAGE"
+  execute gcloud compute instances create $VM_NAME $GCP_OPTS --machine-type $MACHINE_TYPE --image $VM_IMAGE
+}
+
+function run_on_vm() {
+  echo "*** Remote run: \"$1\""
+  execute gcloud compute ssh $VM_NAME $GCP_OPTS --command "$1"
+}
+
+function setup_vm() {
+  run_on_vm 'sudo add-apt-repository ppa:gophers/archive > /dev/null && sudo apt-get update > /dev/null && sudo apt-get install --no-install-recommends -y golang-1.8-go && mv .bashrc .bashrc.orig && (echo "export PATH=/usr/lib/go-1.8/bin:\$PATH:~/go/bin"; cat .bashrc.orig) > ~/.bashrc'
```

Comment:
we use go 1.9 everywhere. Fortio should use it too.

Reply:
oh, I didn't know, when did we switch ? have we already made 1.8 incompatible changes ? 
would be good I compare 1.8 and 1.9 before we do switch
(and I had been staying at 1.8.x for fortio to be compatible with the rest of istio)

Reply:
If you want to ensure that the source remains 1.8 compatible, you can do that. But still use 1.9 compiler.